### PR TITLE
avoid extend query deadline to implement a correct timeout

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Avoid extending deadline when executing query, which satisfy the `query_timeout` semantic.

### CODE REVIEW

**Review by ZiheLiu — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by trueeyu** [comment]
pay attention to the check of tscancode

```
            if (common_metrics != nullptr) {
                common_metrics->remove_counters(std::set<std::string>{"OperatorTotalTime"});
            }

            common_metrics->reset_parent(); // why this don't check != nullptr?
```

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2

File: be/src/exec/pipeline/pipeline_driver_executor.cpp [hunk 1]
```diff
@@ -84,8 +84,6 @@ void GlobalDriverExecutor::_worker_thread() {
                 continue;
             }
 
-            // query context has ready drivers to run, so extend its lifetime.
-            query_ctx->extend_lifetime();
```
Comment (satanson) [inline be/src/exec/pipeline/pipeline_driver_executor.cpp:88]: in QueryContextManager::remove function, extend_lifetime invocation also needs to be removed.

**Comment by wangruin** [comment]
run starrocks_tscancode

**Comment by wangruin** [comment]
run starrocks_tscancode image

File: be/src/exec/pipeline/pipeline_driver_executor.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/pipeline/pipeline_driver_executor.cpp:88]: What's the case?

**Review by satanson — APPROVED** [review]

**Pull request merged**